apollo-compiler/src/test/graphql/com/example/simple_inline_fragment/TestQuery.java

```diff
@@ -335,10 +323,15 @@ public AsDroid read(String conditionalType, ResponseReader reader) throws IOExce
 
         @Override
         public Hero map(ResponseReader reader) throws IOException {
-          final String name = reader.read(fields[0]);
-          final AsHuman asHuman = reader.read(fields[1]);
-          final AsDroid asDroid = reader.read(fields[2]);
-          return new Hero(name, asHuman, asDroid);
+          final AsHuman asHuman = reader.read(fields[0]);
```

Comment:
Yes, that's what Martijn mentioned in slack, we don't support several inline fragments to be acted on Droid and on Character at the same time, for now.

>I don’t think we need to complicate it right now, and we should probably just rely on only one inline fragment matching. I suspect we’ll want to keep using subclasses for concrete objects even if we add better support for interfaces later on, and then we could have the classes implement the interface.